lib/workers/repository/init/inherited.ts

```diff
@@ -99,5 +101,35 @@ export async function mergeInheritedConfig(
       'Removed global config from inherited config.',
     );
   }
-  return mergeChildConfig(config, filteredConfig);
+
+  const decryptedConfig = await decryptConfig(
+    filteredConfig,
+    config.repository,
+  );
+  let returnConfig = decryptedConfig;
+  if (is.nonEmptyArray(returnConfig.extends)) {
+    // Decrypt after resolving in case the preset contains npm authentication instead
+    logger.debug('Resolving presets found in inherited config');
+    returnConfig = await decryptConfig(
+      await resolveConfigPresets(returnConfig, config, config.ignorePresets),
+      config.repository,
+    );
+    logger.trace({ config: returnConfig }, 'Resolved inherited config');
+    res = await validateConfig('inherit', returnConfig);
```

Comment:
You should create a new variable here, for better readability. That way you need no `let` above